### PR TITLE
tests: add %NOLISTENPORT and use it

### PR DESCRIPTION
**Pull request by bagder**
The purpose with this variable is to provide a port number that is
reasonably likely to not have a listener on the local host so that tests
can try connect failures against it. It uses port 2.

Updated six tests to use it instead of the previous different ports.

### CODE REVIEW

File: tests/runtests.pl [hunk 1]
```diff
@@ -126,6 +126,7 @@ BEGIN
 
 my $noport="[not running]";
 
+my $NOLISTENPORT=2;      # port number we use for a local non-listening service
```
Comment (emilengler) [inline tests/runtests.pl:129]: I would suggest using port 47. According to [this](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers), this port is officially reserved.

Comment (bagder) [reply tests/runtests.pl:129]: Yeah, might be a better port. Of course [IANA having it marked as reserved](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt) is not any guarantee that nobody uses it for anything.

I'm also thinking we could have runtests probe the port without the use of curl code to verify it before use.

Comment (emilengler) [reply tests/runtests.pl:129]: Sure, the IANA recommendation is just recommendation. However 47 is reserved where 2 is not so I think it is much more likely that 47 will not be in use.

> I'm also thinking we could have runtests probe the port without the use of curl code to verify it before use.

My only concern with this is the cross-platform compatibility

Comment (bagder) [reply tests/runtests.pl:129]: We already use port 2 successfully, where as we've never tried port 47...

**Comment by bagder** [comment]
thanks @emilengler

**Pull request closed without merging**